packages/core-components/src/layout/Sidebar/Items.tsx

```diff
@@ -83,134 +80,122 @@ export type SidebarItemClassKey =
+      overflow: 'hidden',
+      'text-overflow': 'ellipsis',
+    },
+    iconContainer: props => ({
+      boxSizing: 'border-box',
+      height: '100%',
+      width: props.sidebarConfig.iconContainerWidth,
+      marginRight: -theme.spacing(2),
+      display: 'flex',
+      alignItems: 'center',
+      justifyContent: 'center',
+    }),
+    searchRoot: {
+      marginBottom: 12,
+    },
+    searchField: {
+      color: '#b5b5b5',
+      fontWeight: 'bold',
+      fontSize: theme.typography.fontSize,
+    },
+    searchFieldHTMLInput: {
+      padding: theme.spacing(2, 0, 2),
+    },
+    searchContainer: props => ({
+      width:
+        props.sidebarConfig.drawerWidthOpen -
+        props.sidebarConfig.iconContainerWidth,
+    }),
+    secondaryAction: {
+      width: theme.spacing(6),
+      textAlign: 'center',
+      marginRight: theme.spacing(1),
+    },
+    closedItemIcon: {
+      width: '100%',
+      justifyContent: 'center',
+    },
+    submenuArrow: {
+      display: 'flex',
+    },
+    expandButton: {
+      background: 'none',
+      border: 'none',
+      color: theme.palette.navigation.color,
+      width: '100%',
+      cursor: 'pointer',
+      position: 'relative',
+      height: 48,
+    },
+    arrows: {
+      position: 'absolute',
+      right: 10,
+    },
+    selected: props => ({
+      '&$root': {
+        borderLeft: `solid ${props.sidebarConfig.selectedIndicatorWidth}px ${theme.palette.navigation.indicator}`,
+        color: theme.palette.navigation.selectedColor,
       },
-      expandButton: {
-        background: 'none',
-        border: 'none',
-        color: theme.palette.navigation.color,
-        width: '100%',
-        cursor: 'pointer',
-        position: 'relative',
-        height: 48,
+      '&$closed': {
+        width: props.sidebarConfig.drawerWidthClosed,
       },
-      arrows: {
-        position: 'absolute',
-        right: 10,
+      '& $closedItemIcon': {
+        paddingRight: props.sidebarConfig.selectedIndicatorWidth,
       },
-      selected: {
-        '&$root': {
-          borderLeft: `solid ${sidebarConfig.selectedIndicatorWidth}px ${theme.palette.navigation.indicator}`,
-          color: theme.palette.navigation.selectedColor,
-        },
-        '&$closed': {
-          width: sidebarConfig.drawerWidthClosed,
-        },
-        '& $closedItemIcon': {
-          paddingRight: sidebarConfig.selectedIndicatorWidth,
-        },
-        '& $iconContainer': {
-          marginLeft: -sidebarConfig.selectedIndicatorWidth,
-        },
+      '& $iconContainer': {
+        marginLeft: -props.sidebarConfig.selectedIndicatorWidth,
       },
     }),
-    { name: 'BackstageSidebarItem' },
-  );
-
-// This is a workaround for this issue https://github.com/mui/material-ui/issues/15511
-// The styling of the `selected` elements doesn't work as expected when using a prop callback.
-// Don't use this pattern unless needed
-function useMemoStyles(sidebarConfig: SidebarConfig) {
```

Comment:
Are these changes really intentional? We recently added this as a mitigation to a bug in mui4

Reply:
No, I don't understand why these changes rolled in my commit, The changes I intended to be merged are lines 344-355 and 365 only

Reply:
Hi jhaals, I have fixed the unwanted roll ins, to only the needed intentional changes, Thanks